compiler/ml/typecore.ml

```diff
@@ -3550,11 +3552,15 @@ and type_application ?type_clash_context total_app env funct (sargs : sargs) :
       type_unknown_args max_arity ~args ~top_arity:None omitted ty_fun []
     | (l1, sarg1) :: sargl ->
       let l1 = to_noloc l1 in
+      (* let lbl_name = label_name l1 in
+      Printf.eprintf "  type_unknown_args: lbl_name:%s\n" lbl_name; *)
       let ty1, ty2 =
         let ty_fun = expand_head env ty_fun in
         let arity_ok = List.length args < max_arity in
         match ty_fun.desc with
-        | Tvar _ ->
+        | Tvar _ when (* l1 = Nolabel || *) force_tvar ->
```

Comment:
This is the only change.
Obviously there's some cleanup to do -- but interested in checking that the behaviour is solid for now (and might need to re-introduce debugging code in case issues are found).